### ISSUE TITLE
`Vararg.hastypevars` is false, but issubtype behaves otherwise

### ISSUE THREAD

**Issue by TotalVerb** [JuliaLang/julia#18450]
This is suspicious:

```
julia> Type{Tuple{Vararg{Int}}} <: Type{Tuple{Vararg}}
true

julia> Vararg.hastypevars
false
```

While `!Vararg.hastypevars`, `issubtype` is comparing as if it did.

Example of incorrect behaviour resulting:

``` julia
julia> f() = ifelse(true, Tuple{Vararg{Int}}, Tuple{Vararg})
f (generic function with 1 method)

julia> f()
Tuple{Vararg}

julia> f() == Tuple{Vararg{Int}}
false
```

**Comment by TotalVerb** [comment]
Another case (which I missed with #18467):

```julia
f() = ifelse(true, Tuple{Int, Vararg{Int}}, Tuple{Int, Vararg})
```

**Comment by JeffBezanson** [comment]
closed by #18457
